src/Governor/GovernorGuardian.sol

```diff
@@ -0,0 +1,172 @@
+    /// @notice The cooldown period before a vault can be paused again.
+    uint256 public immutable PAUSE_COOLDOWN;
+
+    /// @notice Event emitted when a vault is paused.
+    /// @param vault The vault that was paused.
+    event Paused(address indexed vault);
+
+    /// @notice Event emitted when a vault is unpaused.
+    /// @param vault The vault that was unpaused.
+    event Unpaused(address indexed vault);
+
+    /// @notice Event emitted when the pause status of a vault changes.
+    /// @param vault The address of the vault whose pause status changed.
+    event PauseStatusChanged(address indexed vault);
+
+    /// @notice Constructor to initialize the contract with the given admin, pause duration, and pause cooldown.
+    /// @param admin The address of the initial admin.
+    /// @param pauseDuration The duration for which the vault remains paused.
+    /// @param pauseCooldown The cooldown period before the vault can be paused again.
+    constructor(address admin, uint256 pauseDuration, uint256 pauseCooldown) {
+        require(pauseDuration > 0 && pauseCooldown > pauseDuration, "constructor error");
+
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+        PAUSE_DURATION = pauseDuration;
+        PAUSE_COOLDOWN = pauseCooldown;
+    }
+
+    /// @notice Executes a call to a specified vault.
+    /// @param vault The address of the vault to call.
+    /// @param data The calldata to be called on the vault.
+    /// @return result The result of the call.
+    function adminCall(address vault, bytes calldata data)
+        external
+        onlyRole(DEFAULT_ADMIN_ROLE)
+        returns (bytes memory)
+    {
+        (bool success, bytes memory result) = vault.call(data);
+        if (!success) RevertBytes.revertBytes(result);
+
+        // If the call is a setHookConfig call, update the pause data.
+        if (bytes4(data) == IEVault(vault).setHookConfig.selector) {
+            (pauseDatas[vault].hookTarget, pauseDatas[vault].hookedOps) = IEVault(vault).hookConfig();
+        }
+
+        return result;
+    }
+
+    /// @notice Pauses the given vaults.
+    /// @param vaults The array of vault addresses to be paused.
+    function pause(address[] calldata vaults) external nonReentrant onlyRole(GUARDIAN_ROLE) {
+        for (uint256 i = 0; i < vaults.length; ++i) {
+            address vault = vaults[i];
+
+            if (!canBePaused(vault)) continue;
+
+            // Cache the hook configuration.
+            (pauseDatas[vault].hookTarget, pauseDatas[vault].hookedOps) = IEVault(vault).hookConfig();
+            pauseDatas[vault].lastPauseTimestamp = uint48(block.timestamp);
+
+            // Disable all operations.
+            IEVault(vault).setHookConfig(address(0), (OP_MAX_VALUE - 1));
+
+            emit Paused(vault);
+        }
+    }
+
+    /// @notice Unpauses the given vaults.
+    /// @param vaults The array of vault addresses to be unpaused.
+    function unpause(address[] calldata vaults) external nonReentrant {
```

Comment:
Maybe there should be a `paused` state similar to the hook target? This would prevent some weird flows like unpausing an already unpaused vault (which would normally be OK, but would overwrite any legit hook changes the governor made in between).

Reply:
Unpausing an already unpaused vault shouldn't be an issue because if the governor made any legit hook changes in between, they would have been cached by the `adminCall` function

Reply:
Ahhhh got it. I missed that bit in `adminCall()`. :+1: